src/Features/Core/Portable/Diagnostics/EngineV2/DiagnosticIncrementalAnalyzer.ProjectState.cs

```diff
@@ -420,14 +442,14 @@ private Task<StrongBox<ImmutableArray<DiagnosticData>>> DeserializeAsync(Diagnos
                     if (entry.Diagnostics.Length == 0)
                     {
                         // if there is no result, use cached task
-                        return s_emptyResultTaskCache;
+                        return s_emptyResultTaskCache!;
                     }
 
-                    return Task.FromResult(new StrongBox<ImmutableArray<DiagnosticData>>(entry.Diagnostics));
+                    return Task.FromResult(new StrongBox<ImmutableArray<DiagnosticData>>(entry.Diagnostics))!;
```

Comment:
❕ Use explicit type arguments, or add an `AsNullable` extension method like the one defined in #38261

Reply:
I'm gonna replace using the StrongBox with ValueTask instead.